apps/codeforafrica/src/lib/data/utils/posts.js

```diff
@@ -0,0 +1,185 @@
+import formatDate from "@/codeforafrica/utils/formatDate";
+
+export function formatTags(tags) {
+  const excludedTags = new Set(["stories", "opportunities"]);
+
+  const tagCounts = tags.reduce((counts, { name }) => {
+    // eslint-disable-next-line no-param-reassign
+    counts[name] = (counts[name] || 0) + 1;
+    return counts;
+  }, {});
+
+  const sortedTags = Object.keys(tagCounts)
+    .filter((tag) => !excludedTags.has(tag.toLowerCase()))
+    .sort((a, b) => tagCounts[b] - tagCounts[a])
+    .map((tagName) => {
+      const tag = tags.find(({ name }) => name === tagName);
+      return {
+        name: tag.name,
+        slug: tag.slug,
+      };
+    });
+
+  return sortedTags;
+}
+
+export function formatPost(post, primaryTag) {
+  const {
+    id,
+    title,
+    coverImage: { src, alt },
+    excerpt,
+    slug,
+    publishedOn,
+    tags,
+  } = post;
+  if (!title) {
+    return null;
+  }
+  return {
+    id,
+    title,
+    image: { src, alt },
+    excerpt,
+    publishedOn: formatDate(publishedOn, {
+      includeTime: false,
+      month: "short",
+    }),
+    tags: formatTags(tags),
+    href: `/posts/${primaryTag}/${slug}`,
```

Comment:
The nested afterRead hook can do this, no?

Reply:
@kilemensi This is the issue I mentioned in the meeting we need to force the link to three slugs.